packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -1136,7 +1136,16 @@ class _ScaffoldLayout extends MultiChildLayoutDelegate {
       }
 
       final double snackBarYOffsetBase;
-      if (floatingActionButtonRect.size != Size.zero && isSnackBarFloating) {
+      final bool showAboveFab = switch (currentFloatingActionButtonLocation) {
+        FloatingActionButtonLocation.startTop => false,
+        FloatingActionButtonLocation.centerTop => false,
+        FloatingActionButtonLocation.endTop => false,
+        FloatingActionButtonLocation.miniStartTop => false,
+        FloatingActionButtonLocation.miniCenterTop => false,
+        FloatingActionButtonLocation.miniEndTop => false,
+        _ => true,
```

Comment:
Instead of a default case, can you enumerate the other values? That way if we ever add another case, we won't introduce a bug where it isn't accounted for. This is actually why we prefer to use switch statements with enums :)

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-if-chains-or--or--with-enum-values

Reply:
> This is actually why we prefer to use switch statements with enums :)
I'm not used to rely on switch, thanks for your explanations :+1: . 

In this particular case, it turned out that `FloatingActionButtonLocation` is not an enum (values are constants defined in an abstract class and they are initialized by instantiating various subclasses). I think we can't avoid adding a default case because this is not a real enum?
I pushed a commit where I enumerated all values (there are a lot ot them). It seems that this an edge case where maybe we can avoid listing all values for readability?

Reply:
Oh wow I did not realize it was an enum! TIL something new! In that case (and sorry for not realizing it sooner) I would recommend throwing an error for the default case. That way, if we add a new location, there will be an error that let's us know we need to account for it here. :)